File: src/main/java/tn/esprit/devdream/entities/Tache.java
package tn.esprit.devdream.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name="Tache")

public class Tache implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id_tache")
    private Long id_tache;
    private Integer priorite;
    private String Description;

    @Enumerated(EnumType.STRING)
    private Tache_status Status;

    private String delai;
    private String performance;
    private String remarque;

    @OneToMany(mappedBy = "tache",cascade = {CascadeType.MERGE,CascadeType.PERSIST})
    private List<Commentaire> commentaireList;

    @OneToMany(mappedBy = "tache",cascade = {CascadeType.MERGE,CascadeType.PERSIST})
    private List<Etiquette> etiquetteList;

    @ManyToOne
            @JoinColumn(name="encadrant_id")
    private User encadrant;

    @ManyToOne
    @JoinColumn(name="etudiant_id")
    private User etudiant;



}

File: src/app/Models/tache.ts
import { Commentaire } from './commentaire';
import { Etiquette } from './etiquette';
import { User } from './user';


export class Tache {
    id_tache?: number;
    priorite: number;
    description: string;
    status: Tache_status;
    delai: string;
    performance: string;
    remarque: string;
    commentaireList?: Commentaire[];
    etiquetteList?: Etiquette[];
    encadrant?: User;
    etudiant?: User;
}
export enum Tache_status {
    Todo = 'Todo',
    InProgress = 'InProgress',
    Done = 'Done'
  }
